craftbook-bukkit/src/main/java/org/enginehub/craftbook/mechanics/Elevator.java

```diff
@@ -218,8 +218,14 @@ public void onCommonClick(PlayerInteractEvent event) {
             case BOTH:
                 if (event.getInteractionPoint() != null) {
                     double relativeHeight = event.getInteractionPoint().getY();
-                    relativeHeight -= (int) relativeHeight;
-                    shift = relativeHeight >= 0.5 ? BlockFace.UP : BlockFace.DOWN;
+                    double fractionalPart = relativeHeight - Math.floor(relativeHeight);  // Normalize to 0.0-1.0
```

Comment:
Couldn’t this be done by just making relativeHeight the absolute value of Y initially? That feels cleaner than having to have two separate code paths where one needs to normalise the range

Reply:
I have tested [the version](https://github.com/RoyalDecree/CraftBook/commit/d3e7b2439f8e56e667b4cfe8ef933f1266b17bbc) 
```java
relativeHeight = Math.abs(relativeHeight % 1.0);
```

where the absolute value of Y in Negative space have caused an unintentional results:

- The UpDown Sign works correctly in positive Y conditions
- When the Y is Negative, the absolute value of interaction point  is also inverted (the bottom half become the top half)
- - this behavior can also be confirmed by walking on non-full blocks like farmland in negative y-space (+y.06250 instead of -y.93750) your position is not subtracted but added to the absolute value so absolute value cannot be used
- so the workaround for negative Y conditions was there to prevent unintentional results where right clicking top half takes you down a floor

Reply:
fair, i'll merge in the current one- thanks :)